extension/src/test/suite/AutoIndent.test.ts

```diff
@@ -0,0 +1,292 @@
+import * as chai from "chai";
+import * as vscode from "vscode";
+import * as fs from "fs";
+import * as path from "path";
+import {findContainers,getIndentation,isCursorInDoubleQuoteExpr} from "../../format/autoIndent";
+import { ReadonlyDocument } from "../../project/readOnlyDocument";
+let assert = chai.assert;
+const testDir = path.join(__dirname + "/../../../extension/src/test");
+const outputDir = path.join(testDir + "/OutputFile");
+let indentTestFile = "";
+let indentTestDoc: vscode.TextDocument;
+
+fs.mkdir(outputDir, { recursive: true }, (err) => {
+  if (err) {
+    return console.error(err);
+  }
+});
+
+function createFakeTextDcoument() {
+  indentTestFile = path.join(testDir + "/SourceFile/autoIndentTestFile.lsp");
+  try {
+    if (fs.existsSync(indentTestFile)) {
+      indentTestDoc = ReadonlyDocument.open(indentTestFile);
+    }
+  } catch (error) {
+    console.log(error);
+  }
+}
+function getIndent(cursorPos2d: vscode.Position) {
+  try {
+    let containerElement = findContainers(indentTestDoc,cursorPos2d);
+    let lineIndentSpace = getIndentation(indentTestDoc,containerElement,cursorPos2d);
+    return lineIndentSpace.length;
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+suite("Autoindent Tests", function () {
+  // Autoindent is used for input in the vscode editor
+  // Doesn't include the case that press enter insider brackets since
+  // the limitation of the unit test
+  // In the client the OnTypeFormattingEdit is trigged after enter so the cursor position is from new line.
+  suiteSetup(() => {
+    createFakeTextDcoument();
+  });
+
+  test("Autoindent for defun enter should have 2 indentation", async function () {
+    let fn = "defun enter";
+    try {
+      let cursorPos2d = new vscode.Position(0, 28);
+      let lineIndentSpace = getIndent(cursorPos2d);
+      assert.isTrue(lineIndentSpace == 2);
+    } catch (err) {
+      assert.fail(`Autoindent test for ${fn} failed`);
+    }
+  });
+
+  test("Autoindent for setq enter should have 2 indentation", async function () {
+    let fn = "setq enter";
```

Comment:
why can't we just use plain text in the call to assert.fail?

Reply:
Do you mean to open a .txt file ?

Reply:
I mean instead of "assert.fail(`Autoindent test for ${fn} failed`);"
you could just use "assert.fail(`Autoindent test for setq enter failed`);"

Reply:
This will reduce the work of hard copy the message. I just need to copy this sentense without modification for every test cases.

Reply:
No it will not reduce the work, as you then will have to change this code:
 let fn = "setq enter";

Reply:
This is a keyword for this test case so I want to put this here in front of the test.